### PR TITLE
REL: update Paver script for sdist completeness and auto-switch test warnings

### PR DESCRIPTION
**Pull request by rgommers**
See gh-6461 for discussion on automatically switching `NoseTester` from `'develop'` to `release`, and https://github.com/scipy/scipy/issues/5195 for what the `paver sdist` change fixes.

### CODE REVIEW

File: pavement.py [hunk 1]
```diff
@@ -549,8 +549,16 @@ def tarball_name(type='gztar'):
 
 @task
 def sdist(options):
+    # First clean the repo and update submodules (for up-to-date doc html theme
+    # and Sphinx extensions)
+    sh('git clean -xdf')
+    sh('git submodule init')
+    sh('git submodule update')
```
Comment (charris) [inline pavement.py]: Hmm, does this need `--force`?

**Comment by charris** [comment]
LGTM, just one question.

File: pavement.py (hunk 1 above)
Comment (rgommers) [inline pavement.py]: I've certainly never needed that. Are you assuming that you may have local changes to the submodule code? That seems far less likely than changes to normal numpy code, which we also don't check for.

**Comment by charris** [comment]
Thanks Ralf. I tested a bit and it seems that `--force` is not needed. IIRC, there was a case mentioned several years back, but I don't remember the specifics.

**Pull request merged**